central/networkgraph/entity/datastore/datastore_impl_sac_test.go

```diff
@@ -0,0 +1,760 @@
+			expectedFound: false,
+		},
+		"Cluster partial access cannot get cluster entity": {
+			contextName:   sacTestUtils.Cluster1NamespaceAReadWriteCtx,
+			entityID:      cluster1EntityID.String(),
+			expectedFound: false,
+		},
+		"Cluster partial access cannot get missing entity": {
+			contextName:   sacTestUtils.Cluster1NamespaceAReadWriteCtx,
+			entityID:      missingEntityID.String(),
+			expectedFound: false,
+		},
+		"Other cluster access cannot get global entity": {
+			contextName:   sacTestUtils.Cluster2ReadWriteCtx,
+			entityID:      globalEntityID.String(),
+			expectedFound: false,
+		},
+		"Other cluster access cannot get cluster entity": {
+			contextName:   sacTestUtils.Cluster2ReadWriteCtx,
+			entityID:      cluster1EntityID.String(),
+			expectedFound: false,
+		},
+		"Other cluster access cannot get missing entity": {
+			contextName:   sacTestUtils.Cluster2ReadWriteCtx,
+			entityID:      missingEntityID.String(),
+			expectedFound: false,
+		},
+	}
+}
+
+type listTestCase struct {
+	contextName      string
+	expectedIDs      []string
+	expectedEntities []*storage.NetworkEntity
+}
+
+func getListTestCases(t testing.TB) map[string]listTestCase {
+	globalEntityID := getGlobalEntityID(t)
+	cluster1EntityID := getCluster1EntityID(t)
+
+	allEntityIDs := []string{globalEntityID.String(), cluster1EntityID.String()}
+	clusterEntityIDs := []string{cluster1EntityID.String()}
+	noEntityIDs := make([]string, 0)
+
+	globalEntity := getGlobalEntity(t)
+	cluster1Entity := getCluster1Entity(t)
+
+	allEntities := []*storage.NetworkEntity{globalEntity, cluster1Entity}
+	clusterEntities := []*storage.NetworkEntity{cluster1Entity}
+	noEntities := make([]*storage.NetworkEntity, 0)
+
+	return map[string]listTestCase{
+		"All access can get global and target cluster entities": {
+			contextName:      sacTestUtils.UnrestrictedReadWriteCtx,
+			expectedIDs:      allEntityIDs,
+			expectedEntities: allEntities,
+		},
+		"Full read access can get global and target cluster entities": {
+			contextName:      sacTestUtils.UnrestrictedReadCtx,
+			expectedIDs:      allEntityIDs,
+			expectedEntities: allEntities,
+		},
+		"Cluster full access can only get entities for the target cluster": {
+			contextName:      sacTestUtils.Cluster1ReadWriteCtx,
+			expectedIDs:      clusterEntityIDs,
+			expectedEntities: clusterEntities,
+		},
+		"Cluster partial access cannot get global nor target cluster entities": {
+			contextName:      sacTestUtils.Cluster1NamespaceAReadWriteCtx,
+			expectedIDs:      noEntityIDs,
+			expectedEntities: noEntities,
+		},
+		"Other cluster access cannot get global nor target cluster entities": {
+			contextName:      sacTestUtils.Cluster2ReadWriteCtx,
+			expectedIDs:      noEntityIDs,
+			expectedEntities: noEntities,
+		},
+	}
+}
+
+type writeTestCase struct {
+	contextName   string
+	expectedError error
+}
+
+func getGlobalEntityWriteTestCases(_ testing.TB) map[string]writeTestCase {
```

Comment:
since it's in `_test.go` file we don't need test sentinel
```suggestion
func getGlobalEntityWriteTestCases() map[string]writeTestCase {
```

Reply:
It was more for consistency with the other test case getters.
I can remove that in a follow-up.